### DISCUSSION TITLE
compact deployment

### DISCUSSION THREAD

**Discussion by sebheitzmann** [symfony/symfony#47332] [Q&A]
I have a use case for a light php application to deploy on a small device.

The customer use for the moment the slim framework and i am wondering if symfony with some configuration could compete with slim in term of size of the deployement archive. There is also a nb of file contraint.
I don't know for the moment any project that could "compact" the symfony projet before deployment ( remove useless vendor files, maybe use the container file only ?, etc ... )

If any of you could lead me to some ressource on that it could help me ( and maybe others ).

Thanks in advance.

**Comment by lyrixx** [comment]
What is the ideal target? (app size, and number of files)?

---

I made a very quick bench:

```
composer create-project slim/slim-skeleton --prefer-dist
# vs
symfony new symfony 
```

Results:

**Symfony**: 12Mb, 1537 files
**Slim**: 43Mb, 3358 files

---

**Warning**: I chose the smallest Symfony distribution and the cache was nor created. About the cache, there is [an option](https://github.com/symfony/symfony/blob/ddaedd28bd2e4fc22b64567753cf934fe9d68c4c/src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php#L137) to keep it in one file.
**Warning 2**: I did not remove the dev deps, it can make a big difference

With Symfony, you'll have a smaller app by default, but I'll increase while you add more dependencies  (obviously), but if you select them carefully, you'll be able to have a small app.
